ldclient_test.go

```diff
@@ -30,3 +40,90 @@ func TestOfflineModeAlwaysReturnsDefaultValue(t *testing.T) {
 		t.Errorf("Offline mode should return default value, but doesn't")
 	}
 }
+
+type evaluateTestData struct {
+	on            bool
+	deleted       bool
+	featureKey    string
+	expectedValue interface{}
+	expectError   bool
+}
+
+func TestEvaluate(t *testing.T) {
+	testData := []evaluateTestData{
```

Comment:
Cool. This looks reasonable so far but I'll hold off on thumbing until that refactoring.